resources/js/smoothScroll.js

```diff
@@ -0,0 +1,38 @@
+const homeSection = document.getElementById("home");
+const contactSection = document.getElementById("contact");
+const projectsSection = document.getElementById("projects");
+const scrollToContactBtn = document.getElementById("scrollToContactBtn");
+const scrollToProjectsBtn = document.getElementById("scrollToProjectsBtn");
+const backToTopArrow = document.getElementById("backToTopBtn");
+
+function smoothScroll(button, section, offsetValue) {
+    const targetPosition = section.offsetTop - offsetValue;
+    button.addEventListener("click", function () {
+        window.scrollTo({
+            top: targetPosition,
+            behavior: "smooth",
+        });
+    });
+}
+
+function checkScrollPosition() {
+    const scrollPos = window.scrollY || document.documentElement.scrollTop;
+    console.log(backToTopArrow.classList.contains("hidden"));
+
+    if (scrollPos >= 300 && backToTopArrow.classList.contains("hidden")) {
+        backToTopArrow.classList.remove("hidden");
+    } else if (
+        scrollPos < 300 &&
+        !backToTopArrow.classList.contains("hidden")
+    ) {
+        backToTopArrow.classList.add("hidden");
+    }
+}
+
+// Show/hide the button based on scroll position
+checkScrollPosition();
+window.addEventListener("scroll", checkScrollPosition);
+
+smoothScroll(backToTopArrow, homeSection, 68);
+smoothScroll(scrollToContactBtn, contactSection, 75);
+smoothScroll(scrollToProjectsBtn, projectsSection, 50);
```

Comment:
les variables sont déclarées pour être utilisées qu'une fois. Du coup on peut les enlever

```
const homeSection = ;
const contactSection = ;
const projectsSection = ;
```

```suggestion
smoothScroll(backToTopArrow, document.getElementById("home"), 68);
smoothScroll(scrollToContactBtn, document.getElementById("contact"), 75);
smoothScroll(scrollToProjectsBtn, document.getElementById("projects"), 50);
```